src/hardware/agilent-54621d/protocol.c

```diff
@@ -0,0 +1,1032 @@
+					        ":TRIG:A:PATT:MODE OFF;" \
+					        ":TRIG:A:PATT:SOUR \"%s\"",
+	[SCPI_CMD_GET_HIGH_RESOLUTION]	      = ":ACQ:HRES?",
+	[SCPI_CMD_SET_HIGH_RESOLUTION]	      = ":ACQ:HRES %s",
+	[SCPI_CMD_GET_PEAK_DETECTION]	      = ":ACQ:TYPE?",				//Fixed
+	[SCPI_CMD_SET_PEAK_DETECTION]	      = ":ACQ:TYPE PEAK",			//Fixed
+	[SCPI_CMD_GET_DIG_CHAN_STATE]	      = ":DIG%d:DISP?",				//Fixed
+	[SCPI_CMD_SET_DIG_CHAN_STATE]	      = ":DIG%d:DISP %s",			//Fixed
+	[SCPI_CMD_GET_VERTICAL_OFFSET]	      = ":CHAN%d:OFFS?",			//Fixed
+	[SCPI_CMD_GET_HORIZ_TRIGGERPOS]	      = ":TIM:POS?",				//OK
+	[SCPI_CMD_SET_HORIZ_TRIGGERPOS]	      = ":TIM:POS %s",				//OK
+	[SCPI_CMD_GET_ANALOG_CHAN_STATE]      = ":CHAN%d:DISP?",			//Fixed
+	[SCPI_CMD_SET_ANALOG_CHAN_STATE]      = ":CHAN%d:DISP %s",			//Fixed
+	[SCPI_CMD_GET_PROBE_UNIT]	      	  = ":CHAN%d:UNIT?",			//Fixed
+	[SCPI_CMD_GET_DIG_POD_THRESHOLD]      = ":POD%d:THR?",				//OK
+	[SCPI_CMD_SET_DIG_POD_THRESHOLD]      = ":POD%d:THR %s",			//OK
+};
+
+static const uint32_t devopts[] = {
+	SR_CONF_OSCILLOSCOPE,
+	SR_CONF_SAMPLERATE | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_TIMEBASE | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_NUM_HDIV | SR_CONF_GET,
+	SR_CONF_HORIZ_TRIGGERPOS | SR_CONF_GET | SR_CONF_SET,
+	SR_CONF_TRIGGER_SOURCE | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_TRIGGER_SLOPE | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_TRIGGER_LEVEL | SR_CONF_GET | SR_CONF_SET,
+//	SR_CONF_TRIGGER_PATTERN | SR_CONF_GET | SR_CONF_SET,			//ToDo: Implement pattern trigger
+	SR_CONF_PEAK_DETECTION | SR_CONF_GET | SR_CONF_SET,
+//	SR_CONF_AVERAGING | SR_CONF_GET | SR_CONF_SET,
+	SR_CONF_AVG_SAMPLES | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_DATA_SOURCE | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST, 
+	SR_CONF_LIMIT_SAMPLES | SR_CONF_GET | SR_CONF_SET,				//The device doesn't actually support limiting samples, but will allways capture the maximum available amout of samples. However the driver can selectively transfere a subset of samples inorder to reduce transfer times.
+};
+
+static const uint32_t devopts_cg_analog[] = {
+	SR_CONF_NUM_VDIV | SR_CONF_GET,
+	SR_CONF_VDIV | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_COUPLING | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_ENABLED | SR_CONF_GET | SR_CONF_SET,
+	//SR_CONF_PROBE_FACTOR | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+};
+
+static const uint32_t devopts_cg_digital[] = {
+	SR_CONF_LOGIC_THRESHOLD | SR_CONF_GET | SR_CONF_SET | SR_CONF_LIST,
+	SR_CONF_LOGIC_THRESHOLD_CUSTOM | SR_CONF_GET | SR_CONF_SET,
+	SR_CONF_ENABLED | SR_CONF_GET | SR_CONF_SET,
+};
+
+static const char *coupling_options[] = {
+	"AC",
+	"DC",
+	"GND",
+	//Agilent 5464x Scopes also allow 50Ohm Termination, however this is configured using the termination command. If someone wants to implement support for that you need to figure out the best way to do that 
+};
+
+static const char *scope_trigger_slopes[] = {
```

Comment:
for this and others where order matters, should have a comment stating so (I think you did it in one place, but these look like they may need it too)

Reply:
This should work just fine when swapped. The index to the currently slected is saved and you actually send "POS" and "NEG" to the device. Since "POS" and "NEG" are also shown in PV it is just looked up whenever it is needed.
Should I add another abstraction to show something else in PV?